src/openMVG/multiview/projection.cpp

```diff
@@ -25,29 +25,44 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-#include "openMVG/multiview/projection.hpp"
+#include <openMVG/multiview/projection.hpp>
+
+#include <cmath>
 
 namespace openMVG {
 
 /// Compute P = K[R|t]
-void P_From_KRt(
-  const Mat3 &K,  const Mat3 &R,  const Vec3 &t, Mat34 *P) {
-  *P = K * HStack(R,t);
+void P_From_KRt(const Mat3 &K,  const Mat3 &R,  const Vec3 &t, Mat34 *P)
+{
+  assert(P != nullptr);
+  *P = P_From_KRt(K, R, t);
+}
+
+Mat34 P_From_KRt(const Mat3 &K, const Mat3 &R, const Vec3 &t)
+{
+  return K * HStack(R,t);
 }
 
-void KRt_From_P(const Mat34 &P, Mat3 *Kp, Mat3 *Rp, Vec3 *tp) {
+void KRt_From_P(const Mat34 &P, Mat3 *Kp, Mat3 *Rp, Vec3 *tp)
+{
+  assert(Kp != nullptr);
+  assert(Rp != nullptr);
+  assert(tp != nullptr);
+  
   // Decompose using the RQ decomposition HZ A4.1.1 pag.579.
   Mat3 K = P.block(0, 0, 3, 3);
 
   Mat3 Q;
   Q.setIdentity();
 
   // Set K(2,1) to zero.
-  if (K(2,1) != 0) {
+  if (K(2,1) != 0)
+  {
     double c = -K(2,2);
     double s = K(2,1);
-    double l = sqrt(c * c + s * s);
-    c /= l; s /= l;
+    const double lenght = std::sqrt(c * c + s * s);
```

Comment:
Using this http://en.cppreference.com/w/cpp/numeric/math/hypot would be even better because it takes care of avoiding overflow/underflow in the intermediate stages.